features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Documents/DocumentsTable.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, {Component} from 'react'
+import {Table, Popconfirm} from 'antd';
+
+class DocumentsTable extends Component {
+    constructor(props) {
+        super(props);
+        this.columns = [{
+            title: 'Name',
+            dataIndex: 'name',
+            key: 'name'
+        }, {
+            title: 'Source',
+            dataIndex: 'sourceType',
+            key: 'sourceType'
+        }, {
+            title: 'Actions',
+            dataIndex: 'actions',
+            key: 'actions',
+            render: (text1, record) => (<div>
+                <a href="#" onClick={() => this.props.onEditAPIDocument(record)}>Edit | </a>
```

Comment:
Since you have referred to this.props inside `constructor` method, You would not able to receive consequence props update inside the constructor(since constructor is called once, when creating an element from a component during initialization) so IMHO it would be better to update `columns` attribute in [componentwillreceiveprops](https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops) method